### ISSUE TITLE
COSI Support

### ISSUE THREAD

**Issue by pschichtel** [benfiola/minio-operator-ext#30]
Thanks for this operator, I'll definitely be using this. I found the project while looking for COSI drivers für MinIO, so I'm wondering it if it might be viable to implement the COSI CRDs in addition to your own CRDs? I could imagine having a simple driver that just translates the COSI CRs into your correspondig CRs and keeping them in sync.

**Comment by benfiola** [comment]
Interesting!  I had no idea there was a SIG working towards standardizing this very thing!  I’ll take a look - to your point, this shouldn’t be too much work (as it’d simply translate the COS resources into the _Minio*_ ones).

**Comment by benfiola** [comment]
Sorry for the delay - I'm still looking into this, just had a busy holiday 😓 .

**Comment by pschichtel** [comment]
> I see you found minio-operator-ext through a similar request to MinIO for COSI support - hoping that the operator is working well for you!

yes exactly and yes it's working well for what we need. Unfortunately we will soon migrate away from MinIO and so this operator might become irrelevant for me.

**Comment by benfiola** [comment]
Heh, it seems like migrating away from MinIO was prudent!

I just wrote up a [document](https://github.com/benfiola/minio-operator-ext/issues/53) regarding the future of this project as it pertains to MinIO entering maintenance mode. The tl;dr is that this project is likely going to enter a maintenance mode of its own as I pivot towards writing an S3-compatible, but backend-agnostic, controller that tries to (re-)implement the functionality provided by this project - as it's unclear whether I'll continue to host MinIO in my own personal cluster.

With a backend-agnostic controller, a COSI implementation would make a lot more sense - so this is something I'll think about as I work on this rewrite.
